b2share/modules/records/triggers.py

```diff
@@ -36,6 +37,7 @@
 
 def register_triggers(app):
     before_record_insert.connect(set_record_schema)
+    before_record_insert.connect(index_record)
```

Comment:
Records are not yet validated when `before_record_insert` is fired. Can you use `after_record_insert` instead?